controllers/storagecluster/cephcluster.go

```diff
@@ -989,6 +990,13 @@ func countAndReplicaOf(ds *ocsv1.StorageDeviceSet) (int, int) {
 	return count, replica
 }
 
+func isDeviceSetToBeEncrypted(sc *ocsv1.StorageCluster, ds ocsv1.StorageDeviceSet) bool {
+	if ds.Encrypted != nil {
+		return *ds.Encrypted
+	}
+	return sc.Spec.Encryption.Enable || sc.Spec.Encryption.ClusterWide
```

Comment:
if cluster is already marked for "clusterwide" encryption during day-1, do we still allow to override a particular future deviceSet(s) from being encrypted ?? cc @malayparida2000 @parth-gr

Reply:
Yes, we can change it when we are adding day-2 device sets,

But once encryption is set on a device set we cant change in 4.18

Reply:
ack !

Reply:
cc @SanjalKatiyar only we change for new device class device sets

And not possible for same device class device sets

Reply:
cc @vbnrh FYI.